Lib/test/libregrtest/runtest_mp.py

```diff
@@ -416,6 +416,8 @@ def _get_result(self) -> QueueOutput | None:
             running = get_running(self.workers)
             if running and not self.ns.pgo:
                 self.log('running: %s' % ', '.join(running))
+            elif not any(worker.is_alive() for worker in self.workers):
```

Comment:
What about changing invariant conditions instead of adding edge conditions?

````python

        while any(worker.is_alive() for worker in self.workers):
            if use_faulthandler:
                faulthandler.dump_traceback_later(MAIN_PROCESS_TIMEOUT,
                                                  exit=True)
            # wait for a thread
            try:
                return self.output.get(timeout=timeout)
            except queue.Empty:
                pass
            # display progress
            running = get_running(self.workers)
            if running and not self.ns.pgo:
                self.log('running: %s' % ', '.join(running))
````

Reply:
I prefer @corona10's code. Also, please add a comment explaining why this test is needed with ``bpo-46205: `` prefix in the comment.